examples/java/basic_example/src/main/java/Plugin/JavaPlugin.java

```diff
@@ -0,0 +1,394 @@
+package Plugin;
+
+import qlik.sse.ServerSideExtension;
+
+import io.grpc.Metadata;
+import io.grpc.stub.StreamObserver;
+import static io.grpc.Metadata.BINARY_BYTE_MARSHALLER;
+import static io.grpc.Metadata.ASCII_STRING_MARSHALLER;
+
+import java.util.logging.Level;
+import java.util.logging.Logger;
+import java.util.List;
+import java.util.ArrayList;
+import java.util.Date;
+
+import javax.script.ScriptEngine;
+import javax.script.ScriptEngineManager;
+
+    
+public class JavaPlugin extends qlik.sse.ConnectorGrpc.ConnectorImplBase {
+     
+    private static final Logger logger = Logger.getLogger(JavaPlugin.class.getName());
+    private ThreadLocal<Metadata> metadata = new ThreadLocal<Metadata>();
+    
+    public void setMetadata(Metadata metadata) {
+        this.metadata.set(metadata);
+    }
+    
+     @Override
+     public void getCapabilities(qlik.sse.ServerSideExtension.Empty request,
+         io.grpc.stub.StreamObserver<qlik.sse.ServerSideExtension.Capabilities> responseObserver) {
+         
+         logger.info("getCapabilities called.");
+         
+         ServerSideExtension.Capabilities pluginCapabilities = ServerSideExtension.Capabilities.newBuilder()
+             .setAllowScript(true)
+             .setPluginIdentifier("Qlik java plugin")
+             .setPluginVersion("v1.0.0")
+             .addFunctions(ServerSideExtension.FunctionDefinition.newBuilder()
+                 .setName("HelloWorld")
+                 .setFunctionId(0)
```

Comment:
Define constants for function IDs since they are also switched upon elsewhere in the code.